rskj-core/src/main/java/co/rsk/net/sync/PeersInformation.java

```diff
@@ -106,6 +109,25 @@ public SyncPeerStatus getPeer(Peer peer) {
     }
 
     public Optional<Peer> getBestPeer() {
+        if (percentageOfPeersToConsiderInRandomSelection > 0.0D) {
+            List<Map.Entry<Peer, SyncPeerStatus>> entries = getBestCandidatesStream()
+                    .sorted(this.peerComparator.reversed())
+                    .collect(Collectors.toList());
+
+            if (entries.isEmpty()) {
+                return Optional.empty();
+            }
+
+            int numberOfPeersToConsider = (int) Math.floor(entries.size() * (percentageOfPeersToConsiderInRandomSelection / 100));
```

Comment:
it seems that `numberOfPeersToConsider ` would equal to zero even if we have one peer and, for instance, percentage equals to 40%?

Reply:
Addressed